Content.Server/VendingMachines/VendingMachineSystem.cs

```diff
@@ -91,6 +104,21 @@ private void OnEmagged(EntityUid uid, VendingMachineComponent component, GotEmag
             component.Emagged = true;
             args.Handled = true;
         }
+        
+        private void OnDamage(EntityUid uid, VendingMachineComponent component, DamageChangedEvent args)
+        {
+            if (component.DispenseOnHitChance != null && _random.Prob(component.DispenseOnHitChance.Value))
+                EjectRandom(uid, true, component);
+        }
```

Comment:
this should really be tied to the amount of damage that gets done

Reply:
could even be gigabased and make a damage modifier set for checking against the damage done, so that blunt damage makes it pop out but not slash/pierce/shock etc